File: app/src/main/java/com/amour/shop/adapter/DeliveryDayAdapter.java
//package com.amour.shop.adapter;
//
//import android.content.Context;
//import android.view.LayoutInflater;
//import android.view.ViewGroup;
//
//import androidx.annotation.NonNull;
//import androidx.core.content.ContextCompat;
//import androidx.recyclerview.widget.RecyclerView;
//
//import com.amour.shop.CallBack.DataCallback;
//import com.amour.shop.classes.Constants;
//import com.amour.shop.classes.UtilityApp;
//import com.amour.shop.Models.DeliveryTime;
//import com.amour.shop.R;
//import com.amour.shop.Utils.DateHandler;
//import com.amour.shop.databinding.RowDayItemBinding;
//
//import java.util.List;
//
//public class DeliveryDayAdapter extends RecyclerView.Adapter<DeliveryDayAdapter.ViewHolder> {
//
//    Context context;
//    DataCallback dataCallback;
//    private List<DeliveryTime> deliveryDayList;
//    public int lastIndex = 0;
//
//
//    public DeliveryDayAdapter(Context context, List<DeliveryTime> deliveryDayList, DataCallback dataCallback) {
//        this.deliveryDayList = deliveryDayList;
//        this.context = context;
//        this.dataCallback = dataCallback;
//
//
//    }
//
//
//    @NonNull
//    @Override
//    public ViewHolder onCreateViewHolder(@NonNull ViewGroup viewGroup, final int i) {
//
//        RowDayItemBinding itemView = RowDayItemBinding.inflate(LayoutInflater.from(context), viewGroup, false);
//        return new ViewHolder(itemView);
//    }
//
//    @Override
//    public void onBindViewHolder(@NonNull final ViewHolder viewHolder, final int position) {
//        final DeliveryTime deliveryTimes = deliveryDayList.get(position);
//        String dayName;
//        String monthName="";
//        String lang=UtilityApp.INSTANCE.getLanguage();
//
//
//        String today=DateHandler.GetDateNowString();
//        String day = DateHandler.FormatDate4(deliveryTimes.getDate(), "yyyy-MM-dd", "dd", lang);
//
//
//
//        if (UtilityApp.INSTANCE.getLanguage().equals(Constants.Arabic)) {
//            monthName = DateHandler.FormatDate4(deliveryTimes.getDate(), "yyyy-MM-dd", "MMMM", lang);
//
//            if(today.equals(deliveryTimes.getDate())){
//                dayName=context.getString(R.string.today);
//            }
//            else {
//                dayName = DateHandler.FormatDate4(deliveryTimes.getDate(), "yyyy-MM-dd", "EEEE");
//
//            }
//
//        } else {
//
//            monthName = DateHandler.FormatDate4(deliveryTimes.getDate(), "yyyy-MM-dd", "MMM", lang);
//
//            if(today.equals(deliveryTimes.getDate())){
//                dayName=context.getString(R.string.today);
//            }
//            else {
//                dayName = (DateHandler.FormatDate4(deliveryTimes.getDate(), "yyyy-MM-dd", "EEEE")).substring(0, 3);
//
//            }
//
//
//        }
//
//        viewHolder.binding.dateTxt.setText(day.concat(" "+monthName));
//
//        viewHolder.binding.dayTxt.setText(dayName);
//
//
//
//        if (lastIndex == position) {
//            viewHolder.binding.cardView.setBackground(ContextCompat.getDrawable(context, R.drawable.round_medium_corner_red));
//            viewHolder.binding.dayTxt.setTextColor(ContextCompat.getColor(context, R.color.white));
//            viewHolder.binding.dateTxt.setTextColor(ContextCompat.getColor(context, R.color.white));
//
//
//        } else {
//            viewHolder.binding.cardView.setBackground(ContextCompat.getDrawable(context, R.drawable.round_medium_corner_unselected_gray));
//            viewHolder.binding.dayTxt.setTextColor(ContextCompat.getColor(context, R.color.black));
//            viewHolder.binding.dateTxt.setTextColor(ContextCompat.getColor(context, R.color.black));
//
//        }
//
//
//    }
//
//    @Override
//    public int getItemCount() {
//        return deliveryDayList.size();
//    }
//
//
//
//    class ViewHolder extends RecyclerView.ViewHolder {
//        RowDayItemBinding binding;
//
//        ViewHolder(RowDayItemBinding view) {
//            super(view.getRoot());
//            binding = view;
//
//            itemView.setOnClickListener(view1 -> {
//                DeliveryTime deliveryDay = deliveryDayList.get(getBindingAdapterPosition());
//
//                lastIndex = getBindingAdapterPosition();
//                notifyDataSetChanged();
//
//                if (dataCallback != null) {
//                    dataCallback.dataResult(deliveryDay, "result", true);
//                }
//            });
//
//
//        }
//    }
//
//}

File: app/src/main/java/com/amour/shop/adapter/DeliveryDayAdapter.kt
package com.amour.shop.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import androidx.core.content.ContextCompat
import androidx.recyclerview.widget.RecyclerView
import com.amour.shop.CallBack.DataCallback
import com.amour.shop.Models.DeliveryTime
import com.amour.shop.R
import com.amour.shop.Utils.DateHandler.FormatDate4
import com.amour.shop.Utils.DateHandler.GetDateNowString
import com.amour.shop.classes.Constants
import com.amour.shop.classes.UtilityApp.getLanguage
import com.amour.shop.databinding.RowDayItemBinding


class DeliveryDayAdapter(
    var context: Context,
    private val deliveryDayList: MutableList<DeliveryTime>,
    var dataCallback: DataCallback?
) :
        RecyclerView.Adapter<DeliveryDayAdapter.ViewHolder>() {
    var lastIndex = 0
    override fun onCreateViewHolder(viewGroup: ViewGroup, i: Int): ViewHolder {
        val itemView = RowDayItemBinding.inflate(LayoutInflater.from(context), viewGroup, false)
        return ViewHolder(itemView)
    }

    override fun onBindViewHolder(viewHolder: ViewHolder, position: Int) {
        val deliveryTimes = deliveryDayList[position]
        val dayName: String
        var monthName = ""
        val lang = getLanguage()
        val today = GetDateNowString()
        val day = FormatDate4(deliveryTimes.date, "yyyy-MM-dd", "dd", lang)
        if (getLanguage() == Constants.Arabic) {
            monthName = FormatDate4(deliveryTimes.date, "yyyy-MM-dd", "MMMM", lang)
            dayName = if (today == deliveryTimes.date) {
                context.getString(R.string.today)
            } else {
                FormatDate4(deliveryTimes?.date?:"", "yyyy-MM-dd", "EEEE")
            }
        } else {
            monthName = FormatDate4(deliveryTimes.date, "yyyy-MM-dd", "MMM", lang)
            dayName = if (today == deliveryTimes.date) {
                context.getString(R.string.today)
            } else {
                FormatDate4(deliveryTimes.date, "yyyy-MM-dd", "EEEE").substring(0, 3)
            }
        }
        viewHolder.binding.dateTxt.text = "$day $monthName"
        viewHolder.binding.dayTxt.text = dayName
        if (lastIndex == position) {
            viewHolder.binding.cardView.background =
                ContextCompat.getDrawable(context, R.drawable.round_medium_corner_red)
            viewHolder.binding.dayTxt.setTextColor(ContextCompat.getColor(context, R.color.white))
            viewHolder.binding.dateTxt.setTextColor(ContextCompat.getColor(context, R.color.white))
        } else {
            viewHolder.binding.cardView.background =
                ContextCompat.getDrawable(context, R.drawable.round_medium_corner_unselected_gray)
            viewHolder.binding.dayTxt.setTextColor(ContextCompat.getColor(context, R.color.black))
            viewHolder.binding.dateTxt.setTextColor(ContextCompat.getColor(context, R.color.black))
        }
    }

    override fun getItemCount(): Int {
        return deliveryDayList.size
    }

    inner class ViewHolder(var binding: RowDayItemBinding) : RecyclerView.ViewHolder(
        binding.root
    ) {
        init {
            itemView.setOnClickListener { view1: View? ->
                val deliveryDay = deliveryDayList[bindingAdapterPosition]
                lastIndex = bindingAdapterPosition
                notifyDataSetChanged()
                if (dataCallback != null) {
                    dataCallback!!.dataResult(deliveryDay, "result", true)
                }
            }
        }
    }
}